src/sql/parts/objectExplorer/common/objectExplorerService.ts

```diff
@@ -377,8 +440,17 @@ export class ObjectExplorerService implements IObjectExplorerService {
 
 		let provider = this._providers[providerId];
 		if (provider) {
-			return provider.closeSession({
+			this._topLevelChildrenPath = [];
```

Comment:
This clearing out of topLevelChildrenPath indicates you're using 1 global set of paths even though you can have multiple sessions open - please test with 2 or more sessions open at once to spot the likely issues. In this case, you'll end up accidentally losing info for the other sessions when you close one